rclpy/rclpy/context.py

```diff
@@ -36,21 +40,29 @@ def __init__(self):
         self._lock = threading.Lock()
         self._callbacks = []
         self._callbacks_lock = threading.Lock()
+        self._logging_initialized = False
 
     @property
     def handle(self):
         return self._handle
 
-    def init(self, args: Optional[List[str]] = None):
+    def init(self, args: Optional[List[str]] = None, *, initialize_logging: bool = True):
         """
         Initialize ROS communications for a given context.
 
         :param args: List of command line arguments.
         """
         # imported locally to avoid loading extensions on module import
         from rclpy.impl.implementation_singleton import rclpy_implementation
+        global g_logging_ref_count
         with self._handle as capsule, self._lock:
             rclpy_implementation.rclpy_init(args if args is not None else sys.argv, capsule)
+            if initialize_logging:
+                self._logging_initialized = True
```

Comment:
@ivanpauno should this be after the context manager, in case `rclpy_logging_configure` raises?

Reply:
It shouldn't be after the context manager because in that case in can be initialized twice.
But it definitely has to be after the `rclpy_logging_configure` function call.

Reply:
> It shouldn't be after the context manager because in that case it can be initialized twice.

I'm sorry, this is not correct as the local `lock` is protecting that.

Reply:
Fixed in https://github.com/ros2/rclpy/pull/518/commits/a5814ed5b7c5788ecd6e77aa11fa36580ed8efaa.